hawc/client/__init__.py

```diff
@@ -0,0 +1,134 @@
+from typing import Dict, List, Optional
+
+import pandas as pd
+from requests import Response, Session
+
+
+class HawcClientException(Exception):
+    """An exception occurred in the HAWC client module."""
+
+    pass
+
+
+class HawcServerException(Exception):
+    """An exception occurred on the HAWC server."""
+
+    pass
+
+
+class Client:
+    def __init__(self, email: str, password: str, root_url: str = "https://hawcproject.org"):
+        self.root_url = root_url
+        self.session = self.authenticate(email, password)
+
+    def _handle_hawc_response(self, response: Response) -> Optional[Dict]:
+        """
+        Handle standard hawc API response
+
+        Args:
+            response (Response): A response object
+
+        Raises:
+            HawcClientException: If error occurs in data submission
+            HawcServerException: If error orccurs on server
+
+        Returns:
+            Dict: The JSON response, if exists
+        """
+        if response.status_code >= 400 and response.status_code < 500:
+            raise HawcClientException(response.status_code, response.json())
+        elif response.status_code == 500:
+            raise HawcServerException(response)
+        else:
+            return response.json()
+
+    def authenticate(self, email: str, password: str) -> Session:
+        """
+        Authenticate a user session
+
+        Args:
+            email (str): email to authenticate
+            password (str): password to authenticate
+
+        Returns:
+            Session: a requests.Session object
+        """
+        session = Session()
+
+        url = f"{self.root_url}/user/api/token-auth/"
+        data = {"username": email, "password": password}
+        response = session.post(url, json=data)
+        token = self._handle_hawc_response(response)["token"]
+        print(response)
+        session.headers.update(Authorization=f"Token {token}")
+        return session
+
+    def get(self, url: str) -> Optional[Dict]:
```

Comment:
perhaps create a "private" method to `_format_to_df()` which takes the response we'll get from all of these and returns a data frame.

Also, these two methods are private, aka they shouldn't be exposed to our public api, so prefix them with a single underscore, eg., `_get()`, `_post()`, this is a pretty standard python convention for "private" stuff

Reply:
get and post methods were made private. I managed to use the dataframe constructor directly on the json, so it did not warrant a separate helper method.